### ISSUE TITLE
controller.apiDocumentation() not working as expected

### ISSUE THREAD

**Issue by riverdusty** [arangodb/arangodb#1461]
https://docs.arangodb.com/Foxx/Develop/ApiDocumentation.html
I have tried to emulate the above URL exactly but get an error response:
```
Can't read swagger JSON from http://localhost:8529/_db/MyDB/MyFoxxApp/Test/docs/swagger.json?mount=/_admin/aardvark/foxxes/docu/MyFoxx
```
Using URL:
```
http://localhost:8529/_db/MyDB/MyFoxx/Test/docs/index.html?mount=/_admin/aardvark/foxxes/docu/MyFoxx
```
It works if you leave off the ```/foxxes/docu/MyFoxx``` by then its pretty pointless in trying to return API documentation to a 3rd party application.

You might be thinking I could just use the provided URL, but in my situation, the host and database name may change across clients, so I cannot rely on a static URL and host.

**Comment by pluma** [comment]
Ah, I think the queryParam for mount is missing from the example, meaning req.parameters.mount isn't getting set.

**Comment by pluma** [comment]
I'm a bit confused about what you're specifically trying to do. The example is for creating a mount point that lets you see the API docs of any mounted app by its mount point.

Are you really trying to get your app to show the API docs of another app mounted at `/_admin/aardvark/foxxes/docu/MyFoxx`? That seems like an odd mount point.

**Comment by riverdusty** [comment]
I'm trying to deliver my fox apps own API documentation via rest to an external client.
I thought the above was the way to do that.
So the "Download API Documentation" link from the ArangoDB interface, I'd like to serve that data to external clients via rest.
The external client basically has ```host+port+"/_db/MyDB/MyFoxx/"+request```
I want the returned result (assming ```request``` is to return API documentation) to be the same as the result of hitting the "Download API Documentation" link.

**Comment by pluma** [comment]
If you only want the Foxx apps' own API documentation, you can use the first example:

```js
controller.apiDocumentation('/my/dox');
```

The *Swagger JSON* is then provided at `/my/dox/swagger.json` (relative to the controller and the app).

The "Download API Documentation" feature in ArangoDB 2.6 is currently linking to the original Swagger 1 JSON. The upcoming 2.7 release will instead provide only the Swagger 2 JSON. There is no way way to get the Swagger 1 JSON outside of the web admin interface and there will be no way to get the Swagger 1 JSON at all starting with ArangoDB 2.7.

In other words, the "/_admin/aardvark/foxxes/docu" routes will go away and be replaced by the same routes used to display the API docs everywhere else.

**Comment by riverdusty** [comment]
Thanks! I thought '/my/dox' was some variable, didn't realise I'm supposed actually type that exactly. Its working now, however, the JSON is different from the link on the Foxx ui. No biggy.

**Comment by pluma** [comment]
@riverdusty Great to hear. Could you clarify what you mean by "different"? If you're using ArangoDB 2.6 or earlier (i.e. not the latest development release) that difference is to be expected as the link would show Swagger 1 JSON, otherwise something weird is going on.

**Comment by riverdusty** [comment]
Yeah, I'm on 2.6.2 I think. I'll upgrade and see if its changed at all.
Just checked and ```/my/dox``` is swagger 2 while the link is swagger one.

**Comment by dothebart** [comment]
The API-Documentation now is also on swagger2, will be part of arangodb 2.7
